modules/standard/LinkedLists.chpl

```diff
@@ -203,6 +203,52 @@ record LinkedList {
      return ret;
    }
 
+  /*
+    find `e` in the list.
+
+    :arg e: The element to be checked for
+    :return: True if the item was found, otherwise returns false
+    :rtyle: `bool`
+   */
+  proc contains(e : eltType): bool {
+    for item in this {
+      if (e == item) {
+        return true;
+      }
+    }
+    return false;
+  }
+  
+  /*
+    Returns the data stored in the first element of the list
+    
+    :return: the data stored at the front of the list
+    :rtype: `ref eltType`
+    :throws SystemError: if the list is empty
+   */
+  proc ref front() ref: eltType throws {
```

Comment:
@Liannium I am a bit confused. Why you are using the `ref` keyword in `proc` definition?

I think `proc front(): eltType throws {` would be sufficient

Reply:
The `ref` used in `proc ref front...` tells the compiler that the `front` method modifies the receiver of the call. While it's technically true that nothing is being modified in the definition of the `front` routine, we mark it as `ref` because we are returning a mutable reference to an object contained in the linked list.